src/frontend/js/pages/DashboardOpenEdxProfile/index.tsx

```diff
@@ -111,7 +111,24 @@ export const DEFAULT_DISPLAYED_FORM_VALUE = ' - ';
 const DashboardOpenEdxProfile = () => {
   const intl = useIntl();
   const { user } = useSession();
-  const { data: openEdxProfileData, error } = useOpenEdxProfile({ username: user!.username });
+  const {
+    data: openEdxProfileData,
+    methods: { invalidate },
+    error,
+  } = useOpenEdxProfile({ username: user!.username });
+
+  const onClickModifyProfile = async () => {
+    window.location.replace(`${context.authentication.endpoint}/account/settings`);
+  };
+
+  // Because the user can leaver the page to update his profile in openEdx.
+  // if we invalidate before leaving the page, it's directly refetched.
+  // We need to invalidate openEdx's profile each time we reload this page.
+  // it's the only way to have user data synchronised between richie's frontend
+  // and openEdx's database.
+  useEffect(() => {
+    invalidate();
+  }, []);
```

Comment:
Did you try to trigger invalidation on `beforeunload` event emits by window ?
```suggestion
  const onClickModifyProfile = async () => {
    window.addEventListener('beforeunload', invalidate, { once : true });
    window.location.replace(`${context.authentication.endpoint}/account/settings`);
  };
```

Reply:
I tried multiple new solution including this beforeunload.
This event do the same as other ways to trigger invalidate, it invalidate then refetch.
I tried to use `removeQueries` and/or `resetQueries` that execute when the query "enabled" switch to `false`.
It work without the `location.replace`: i trigger the reset, no refresh happened, i change route then comme back, the data is refetch.
But it didn't work with the `location.replace`, the reset is trigger, i change page, edit the profile, come back to richie home then profile -> it found the data from somewhere and don't trigger a new fetch.